lib/open_food_network/user_balance_calculator.rb

```diff
@@ -6,13 +6,29 @@ def initialize(email, distributor)
     end
 
     def balance
-      -completed_orders.sum(&:outstanding_balance)
+      payment_total - completed_order_total
     end
 
     private
 
+    def completed_order_total
+      completed_orders.sum &:total
+    end
+
+    def payment_total
+      payments.sum &:amount
+    end
+
     def completed_orders
-      Spree::Order.where(distributor_id: @distributor, email: @email).complete.not_state(:canceled)
+      all_orders.complete.not_state(:canceled)
+    end
+
+    def all_orders
+      Spree::Order.where(distributor_id: @distributor, email: @email)
+    end
+
+    def payments
+      Spree::Payment.where(order_id: all_orders, state: "completed")	      
```

Comment:
so you are putting back what was there before, right?
https://github.com/openfoodfoundation/openfoodnetwork/pull/4733/files

to do this correctly using git you should get a fresh branch and run:
git revert baf1ecb4366159f063d69fdc77b54c493799406d
commit this revert and then do your new changes.
that would make the change explicit and easy to follow.
this way we dont know what is revert or what are you changing now.

But I think this is simple enough to get us there, I think I figured it out: you reverted and changed this payments method to use all_orders instead of completed_orders, is this correct explanation? is that all?

Reply:
yeah that's it. Payments are read from all_orders and order totals only from completed orders. This was causing issues on payments received on cancelled orders in the first incarnation.